llvm/lib/Analysis/ScalarEvolution.cpp

```diff
@@ -14934,10 +14936,52 @@ SCEVWrapPredicate::SCEVWrapPredicate(const FoldingSetNodeIDRef ID,
 
 const SCEVAddRecExpr *SCEVWrapPredicate::getExpr() const { return AR; }
 
-bool SCEVWrapPredicate::implies(const SCEVPredicate *N) const {
+bool SCEVWrapPredicate::implies(const SCEVPredicate *N,
+                                ScalarEvolution &SE) const {
   const auto *Op = dyn_cast<SCEVWrapPredicate>(N);
+  if (!Op)
+    return false;
+
+  if (setFlags(Flags, Op->Flags) != Flags)
```

Comment:
You could use intersection here, and prove that flags are weaker than implied.

Reply:
It looks like it the moment flags are either `SCEVWrapPredicate::IncrementNSSW` or `SCEVWrapPredicate::IncrementNUSW` and never both. Left as is for now.